src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -162,22 +166,47 @@ private void EnsureAdditionalFileIsReferenced()
             var refdFile = projectSystem.FindFileInProject(initializedProject, additionalFilePath);
             if (refdFile == null)
             {
-                projectSystem.AddFileToProject(initializedProject, additionalFilePath, Constants.AdditionalFilesItemTypeName);
+                AddAdditionalFileToProject(projectSystem, initializedProject, additionalFilePath);
             }
             else
             {
-                var property = VsShellUtils.FindProperty(refdFile.Properties, Constants.ItemTypePropertyKey);
+                EnsureItemTypeIsCorrect(refdFile);
+            }
+        }
+
+        internal /* for testing */ static void AddAdditionalFileToProject(IProjectSystemHelper projectSystem, Project project, string filePath)
```

Comment:
I've just extracted the existing calls to separate methods with try-catch blocks and customised error messages.
The new methods are static to make testing easier.